tests/jax/components/building/system_init_test.py

```diff
@@ -0,0 +1,471 @@
+    """
+
+    system, builder = system_with_random_agent_networks_no_shared_weights
+    system.on_building_init(builder)
+
+    assert builder.store.agent_net_keys == random_networks_agent_net_keys
+    assert (
+        builder.store.network_sampling_setup == random_networks_network_sampling_setup
+    )
+    assert builder.store.unique_net_keys == random_networks_unique_net_keys
+    assert builder.store.net_keys_to_ids == random_networks_net_keys_to_ids
+
+
+def test_custom_network_sampling_with_shared_weights(
+    system_with_custom_network_sampling_with_shared_weights: Tuple[
+        CustomSamplingSystemInit, Builder
+    ],
+    same_network_agent_net_keys: Dict[str, str],
+    same_network_unique_net_keys: List[str],
+    same_network_net_keys_to_ids: Dict[str, int],
+) -> None:
+    """Test whether system network sampling is instantiated correctly.
+
+    Done for custom network sampling setup with shared weights
+    """
+
+    system, builder = system_with_custom_network_sampling_with_shared_weights
+    system.on_building_init(builder)
+
+    assert builder.store.agent_net_keys == same_network_agent_net_keys
+    assert builder.store.unique_net_keys == same_network_unique_net_keys
+    assert builder.store.net_keys_to_ids == same_network_net_keys_to_ids
+
+
+def test_custom_network_sampling_no_shared_weights(
+    system_with_custom_network_sampling_no_shared_weights: Tuple[
+        CustomSamplingSystemInit, Builder
+    ],
+    random_networks_unique_net_keys: List[str],
+    random_networks_net_keys_to_ids: Dict[str, int],
+) -> None:
+    """Test whether system network sampling is instantiated correctly.
+
+    Done for custom network sampling setup without shared weights
+    """
+
+    system, builder = system_with_custom_network_sampling_no_shared_weights
+    system.on_building_init(builder)
+
+    assert list(builder.store.agent_net_keys.keys()) == builder.store.agents
+    assert set(list(builder.store.agent_net_keys.values())).issubset(
+        random_networks_unique_net_keys
+    )
+    assert builder.store.unique_net_keys == random_networks_unique_net_keys
+    assert builder.store.net_keys_to_ids == random_networks_net_keys_to_ids
+
+
+def test_incorrect_sampling_setup_type(
+    fixed_agent_networks_with_incorrect_sampling_setup_type: Tuple[
+        FixedNetworkSystemInit, Builder
+    ],
+    random_agent_networks_with_incorrect_sampling_setup_type: Tuple[
+        RandomSamplingSystemInit, Builder
+    ],
+    custom_agent_networks_with_incorrect_sampling_setup_type: Tuple[
+        CustomSamplingSystemInit, Builder
+    ],
+) -> None:
+    """Tests all system network sampling instantiation ValueErrors."""
+
+    (
+        fixed_network_system,
+        fixed_network_builder,
+    ) = fixed_agent_networks_with_incorrect_sampling_setup_type
+    with pytest.raises(ValueError):
+        assert fixed_network_system.on_building_init(fixed_network_builder) == ValueError  # type: ignore # noqa: E501
```

Comment:
I don't think you need the assert in here as well as the `with pytest.raises(ValueError):`, ditch the `assert ... == Error` and just use the pytest stuff

Reply:
Might also help with the type issues.

Reply:
Thank you for this catch. Updated :call_me_hand: